vault/resource_aws_secret_backend_role.go

```diff
@@ -37,30 +37,39 @@ func awsSecretBackendRoleResource() *schema.Resource {
 			"policy_arns": {
 				Type:          schema.TypeList,
 				Optional:      true,
-				ConflictsWith: []string{"policy_document", "policy", "policy_arn"},
+				ConflictsWith: []string{"policy_document", "policy", "policy_arn", "role_arns"},
 				Description:   "ARN for an existing IAM policy the role should use.",
 				Elem: &schema.Schema{
 					Type: schema.TypeString,
 				},
 			},
+			"role_arns": {
+				Type:          schema.TypeList,
+				Optional:      true,
+				ConflictsWith: []string{"policy_document", "policy", "policy_arn", "policy_arns"},
+				Description:   "ARNs of the AWS roles the role is allowed to assume. Required when credential_type is assumed_role",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
 			"policy_arn": {
 				Type:          schema.TypeString,
 				Optional:      true,
-				ConflictsWith: []string{"policy_document", "policy", "policy_arns"},
+				ConflictsWith: []string{"policy_document", "policy", "policy_arns", "role_arns"},
 				Description:   "ARN for an existing IAM policy the role should use.",
 				Deprecated:    `Use "policy_arns".`,
 			},
 			"policy_document": {
 				Type:             schema.TypeString,
 				Optional:         true,
-				ConflictsWith:    []string{"policy_arns", "policy_arn", "policy"},
+				ConflictsWith:    []string{"policy_arns", "policy_arn", "policy", "role_arns"},
```

Comment:
Per the documentation, there is no conflict
https://www.vaultproject.io/api/secret/aws/index.html#policy_document
Says it will just act as a filter